Code/Framework/AtomCore/AtomCore/Instance/InstanceDatabase.h

```diff
@@ -387,6 +387,14 @@ namespace AZ
         Data::Instance<Type> InstanceDatabase<Type>::EmplaceInstance(
             const InstanceId& id, const Data::Asset<AssetData>& asset, const AZStd::any* param)
         {
+            // This assert is here to catch any potential non-randomness in our id generation. If it triggers,
+            // there might be a bug / race condition in the id generator. The same assert also occurs *after*
+            // instance creation to help differentiate between a non-random id vs recursive creation of the same id.
+            AZ_Assert(
+                m_database.find(id) == m_database.end(),
+                "Database already contains an instance for this id, possibly a random id generation collision?",
+                id.ToString<AZStd::string>().c_str());
```

Comment:
It is better to use ToFixedString to avoid a heap allocation for the Uuid when the assert occurs.
Also the `.contains` function can be used to determine if the id is in the map container.
```suggestion
            AZ_Assert(
                m_database.contains(id),
                "Database already contains an instance for this id, possibly a random id generation collision?",
                id.ToFixedString().c_str());
```

Reply:
InstanceId (which isn't a UUID but does contain one) doesn't have a ToFixedString() method.

Reply:
Ok, alternative solution is to use the template argument to pass in `Uuid::FixedString`